### PR TITLE
fix(modal): add transparent outline for high contrast themes

### LINKED ISSUE #3880: Modal lacks border to delineate edge of dialog, esp. for Low Vision transformations
## Environment
OS: Most reproducible on Windows, using high contrast (HC) mode
Browser: Firefox most reprodicble (adopts HC)
AT: Can occur with themes for magnifiers such as Zoomtext

## Detailed description
The edge of the modal (both the white main dialog and the blue and black button regions) is  defined by the colour contrast with the grey background, as show in this screenshot
image

In cases where the background content is not sufficiently greyed out OR where due to AT/browser transformations, the background or foreground is transformed, the delineation of the edge that is due entirely to contrasting colour can be lost.

For instance, in the following screenshot, Windows High Contrast mode has caused the background to go black, and reversed the colors of the main dialog area. The buttons have a defined border, but the main dialog region lacks any border, making it difficult for a user both to locate and to understand the context. It is not clear the text floating in the middle of the screen constitutes a dialog. (Where magnification is used, the problem is exacerbated.)
image

Similar issues can occur through some ATs.

> What version of the Carbon Design System are you using?
v10 React http://react.carbondesignsystem.com/iframe.html?id=modal--default

> What did you expect to happen?
I would expect that when the high contrast mode is adopted, the border of the dialog would be visible.

> What happened instead?
Only the button borders are visible.

> What WCAG 2.1 checkpoint does the issue violate?
Not clearly a violation, but obviously a DESIGN consideration. 

## Steps to reproduce the issue

1. From the Start menu, type High Contrast mode, and press Enter
1. Choose High Contrast #1
1. Launch the react carbon modal component in Firefox
1. Press F5 to refresh the browser (if it didn't adopt high contrast colours automatically)

> Please create a reduced test case in CodeSandbox
>
> - Style and vanilla JS:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/components/examples/codesandbox
> - React:
>   https://codesandbox.io/s/github/carbon-design-system/carbon/tree/master/packages/react/examples/codesandbox

## Additional information
The simple solution to this is to have a defined border for the dialog. For designers concerned with making the dialog 'ugly' by adding a black stroke around the outside edge, there are ways of implementing a border that need not affect the untranformed view, but which will show up when users or ATs do things like override the CSS of the page.

### PR DESCRIPTION
**Pull request by emyarod**
Closes #3880

This PR adds a transparent outline to the modal container so that the edge of the modal is visible in high contrast environments

#### Changelog

**New**

- modal container outline and outline offset

#### Testing / Reviewing

Ensure the low contrast modal edge can be seen against the background

Ensure the normal modal views are unchanged

### CODE REVIEW

File: packages/components/src/components/modal/_modal.scss [hunk 1]
```diff
@@ -75,6 +75,8 @@
     width: 100%;
     height: 100%;
     max-height: 100%;
+    outline: 3px solid transparent;
```
Comment (joshblack) [inline packages/components/src/components/modal/_modal.scss:79]: Could you add a quick note to explain the outline additions? Make a comment plus link to issue or something?

**Review by joshblack — APPROVED** [review]

**Review by tw15egan — APPROVED** [review]
LGTM 👍

**Review by aagonzales — APPROVED** [review]
LGTM2!

**Review by laurenmrice — APPROVED** [review]
looks good 👍🏻

**Review by asudoh — APPROVED** [review]
LGTM 👍 - Thanks @emyarod!

**Pull request merged**